httpclient/api_client.go

```diff
@@ -236,6 +237,9 @@ func (c *ApiClient) attempt(
 				return c.failRequest(ctx, "failed during request visitor", err)
 			}
 		}
+		// Set traceparent for distributed tracing.
```

Comment:
Users running the SDK in a server can set up their own Traceparent using visitors. This code correctly doesn't override this. But it is important that this line is applied after the visitors. I would add a note about this to avoid moving this during a refactor, as well as a test to ensure that we don't override traceparent set by the users.